ESP8266WirelessPrintAsync/ESP8266WirelessPrintAsync.ino

```diff
@@ -825,59 +851,55 @@ void ReceiveResponses() {
-        GotValidResponse();       // Warning, this will empty 'serialResponse'
-      }
-      else if (parsePosition(serialResponse)) {
-        telnetSend("MPosition parsed");
-        restartSerialTimeout();   // Some firmware doesn't send busy while homing but just position. It can be used to prevent timeout
-        GotValidResponse();       // Warning, this will empty 'serialResponse'
-      }
-      else if (serialResponse.startsWith("echo:busy")) {
-        telnetSend("Printer is busy, giving it more time");
-        restartSerialTimeout();
-        GotValidResponse();   // Warning, this will empty 'serialResponse'
-      }
-      else if (serialResponse.startsWith("echo: cold extrusion prevented")) {
-        telnetSend("Printer is cold, can't move");
-        // To do: Pause sending gcode, or do something similar
-        GotValidResponse();   // Warning, this will empty 'serialResponse'
-      }
-      else if (serialResponse.startsWith("Error:")) {
-        telnetSend("Error received: ");
-        telnetSend("<" + String(serialResponse) );
-        cancelPrint = true;
-        GotValidResponse();   // Warning, this will empty 'serialResponse'
+
+        unsigned int cmdLen = commandQueue.popAcknowledge().length();     // Go on with next command
+        printerUsedBuffer = max(printerUsedBuffer - cmdLen, 0u);
       }
       else {
-        lineStartPos = serialResponse.length();
-        telnetSend("Unhandled line received:");
-        telnetSend("<" + String(serialResponse) );
+        restartSerialTimeout();     // Something has been received so clear timeout (printer and communication are live
+
+        if (parseTemperatures(serialResponse))
+          responseDetail = "autotemp";
+        else if (parsePosition(serialResponse))
+          responseDetail = "position";
+        else if (serialResponse.startsWith("echo:busy"))
+          responseDetail = "busy";
+        else if (serialResponse.startsWith("echo: cold extrusion prevented")) {
+          // To do: Pause sending gcode, or do something similar
+          responseDetail = "cold extrusion";
+        }
+        else if (serialResponse.startsWith("Error:")) {
+          cancelPrint = true;
+          responseDetail = "ERROR";
+        }
+        else {
+          lineStartPos = serialResponse.length();
+          incompleteResponse = true;
+          responseDetail = "wait more";
+        }
+      }
+
+      telnetSend("<" + serialResponse + " #" + responseDetail + "#");
+      if (!incompleteResponse) {
+        lastReceivedResponse = serialResponse;
+        lineStartPos = 0;
+        serialResponse = "";
       }
     }
   }
 
   if (!commandQueue.isAckEmpty() && (signed)(serialReceiveTimeoutTimer - millis()) <= 0) {  // Command has been lost by printer, buffer has been freed
-    telnetSend("< #TIMEOUT#");
+    telnetSend("<#TIMEOUT#");
```

Comment:
Let's use `<` only for strings received from the printer. As `#TIMEOUT#` is not a string received from the printer, it should not start with `<`